Detectors/ITSMFT/ITS/reconstruction/src/CA/gpu/cuda/Tracker.cu

```diff
@@ -0,0 +1,517 @@
+
+    if (iLayer == 0) {
+
+      GPU::layerTrackletsKernel<<< blocksGrid, threadsPerBlock, 0, streamArray[iLayer].get() >>>(primaryVertexContext.getDeviceContext(),
+          iLayer, primaryVertexContext.getDeviceTracklets()[iLayer].getWeakCopy());
+
+    } else {
+
+      GPU::layerTrackletsKernel<<< blocksGrid, threadsPerBlock, 0, streamArray[iLayer].get() >>>(primaryVertexContext.getDeviceContext(),
+          iLayer, primaryVertexContext.getTempTrackletArray()[iLayer - 1].getWeakCopy());
+    }
+
+    cudaError_t error = cudaGetLastError();
+
+    if (error != cudaSuccess) {
+
+      std::ostringstream errorString { };
+      errorString << "CUDA API returned error [" << cudaGetErrorString(error) << "] (code " << error << ")"
+          << std::endl;
+
+      throw std::runtime_error { errorString.str() };
+    }
+  }
+
+  cudaDeviceSynchronize();
+
+  for (int iLayer { 0 }; iLayer < Constants::ITS::CellsPerRoad; ++iLayer) {
+
+    trackletsNum[iLayer] = primaryVertexContext.getTempTrackletArray()[iLayer].getSizeFromDevice();
+    primaryVertexContext.getDeviceTracklets()[iLayer + 1].resize(trackletsNum[iLayer]);
+
+    cub::DeviceScan::ExclusiveSum(static_cast<void *>(primaryVertexContext.getTempTableArray()[iLayer].get()), tempSize[iLayer],
+        primaryVertexContext.getDeviceTrackletsPerClustersTable()[iLayer].get(),
+        primaryVertexContext.getDeviceTrackletsLookupTable()[iLayer].get(),
+        primaryVertexContext.getClusters()[iLayer + 1].size(), streamArray[iLayer + 1].get());
+
+    dim3 threadsPerBlock { GPU::Utils::Host::getBlockSize(trackletsNum[iLayer]) };
+    dim3 blocksGrid { GPU::Utils::Host::getBlocksGrid(threadsPerBlock, trackletsNum[iLayer]) };
+
+    GPU::sortTrackletsKernel<<< blocksGrid, threadsPerBlock, 0, streamArray[iLayer + 1].get() >>>(primaryVertexContext.getDeviceContext(),
+        iLayer + 1, primaryVertexContext.getTempTrackletArray()[iLayer].getWeakCopy());
+
+    cudaError_t error = cudaGetLastError();
+
+    if (error != cudaSuccess) {
+
+      std::ostringstream errorString { };
+      errorString << "CUDA API returned error [" << cudaGetErrorString(error) << "] (code " << error << ")"
+          << std::endl;
+
+      throw std::runtime_error { errorString.str() };
+    }
+  }
+}
+
+template<>
+void TrackerTraits<true>::computeLayerCells(CA::PrimaryVertexContext& primaryVertexContext)
+{
+  std::array<size_t, Constants::ITS::CellsPerRoad - 1> tempSize;
+  std::array<int, Constants::ITS::CellsPerRoad - 1> trackletsNum;
+  std::array<int, Constants::ITS::CellsPerRoad - 1> cellsNum;
+  std::array<GPU::Stream, Constants::ITS::CellsPerRoad> streamArray;
+
+  for (int iLayer { 0 }; iLayer < Constants::ITS::CellsPerRoad - 1; ++iLayer) {
+
+    tempSize[iLayer] = 0;
+    trackletsNum[iLayer] = primaryVertexContext.getDeviceTracklets()[iLayer + 1].getSizeFromDevice();
+<<<<<<< HEAD
+    primaryVertexContext.getTempCellArray()[iLayer].reset(
```

Comment:
And here

Reply:
Hi, I comment only on this one: these are now fixed. Thanks a lot for spotting this. I was not paying much attention on the CUDA part since I cannot test it inside O2 and these conflicts were here since long.